core/src/main/java/io/questdb/griffin/engine/functions/date/ToTimezoneIntervalFunctionFactory.java

```diff
@@ -0,0 +1,143 @@
+import io.questdb.std.Interval;
+import org.jetbrains.annotations.NotNull;
+
+import static io.questdb.std.datetime.TimeZoneRuleFactory.RESOLUTION_MICROS;
+
+public class ToTimezoneIntervalFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "to_timezone(ΔS)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        Function interval = args.getQuick(0);
+        Function timezone = args.getQuick(1);
+
+        if (timezone.isConstant()) {
+            return getIntervalFunction(argPositions, interval, timezone, 1);
+        } else {
+            return new ToTimezoneIntervalFunctionVar(interval, timezone);
+        }
+    }
+
+    @NotNull
+    static IntervalFunction getIntervalFunction(IntList argPositions, Function interval, Function timezone, int multiplier) throws SqlException {
+        final CharSequence tz = timezone.getStrA(null);
+        if (tz != null) {
+            final int hi = tz.length();
+            final long l = Timestamps.parseOffset(tz, 0, hi);
+            if (l == Long.MIN_VALUE) {
+                try {
+                    return new OffsetIntervalFunctionFromRules(
+                            interval,
+                            TimestampFormatUtils.EN_LOCALE.getZoneRules(
+                                    Numbers.decodeLowInt(TimestampFormatUtils.EN_LOCALE.matchZone(tz, 0, hi)), RESOLUTION_MICROS
+                            ),
+                            multiplier
+                    );
+                } catch (NumericException e) {
+                    Misc.free(interval);
+                    throw SqlException.$(argPositions.getQuick(1), "invalid timezone name");
+                }
+            } else {
+                return new OffsetIntervalFunctionFromOffset(
+                        interval,
+                        multiplier * Numbers.decodeLowInt(l) * Timestamps.MINUTE_MICROS
+                );
+            }
+        }
+        throw SqlException.$(argPositions.getQuick(1), "timezone must not be null");
+    }
+
+    private static class ToTimezoneIntervalFunctionVar extends IntervalFunction implements BinaryFunction {
+        private final Function interval;
+        private final Function timezone;
+
+        public ToTimezoneIntervalFunctionVar(Function interval, Function timezone) {
+            this.interval = interval;
+            this.timezone = timezone;
+        }
+
+        @Override
+        public Interval getInterval(Record rec) {
+            final Interval localInterval = new Interval(interval.getInterval(rec).getLo(), interval.getInterval(rec).getHi());
+            try {
+                final CharSequence tz = timezone.getStrA(rec);

```

Comment:
null or invalid timezone must produce `CairoException`, which can also take SQL position to aid user fixing the issue. Functions have to behave consistently regardless if timezone constant value or not.